src/psyclone/configuration.py

```diff
@@ -757,59 +757,47 @@ def __init__(self, config, section):
-                # Handled in the base class APISpecificConfig
-                pass
-            elif key == "compute_annexed_dofs":
-                try:
-                    # Parse setting for redundant computation over annexed dofs
-                    self._compute_annexed_dofs = section.getboolean(
-                        'COMPUTE_ANNEXED_DOFS')
-                except ValueError as err:
-                    raise ConfigurationError(
-                        "error while parsing COMPUTE_ANNEXED_DOFS in the "
-                        "[dynamo0.3] section of the config file: {0}".
-                        format(str(err)), config=self._config)
-            elif key == "default_kind":
-                # Set default kinds (precisions) from config file
-                all_kinds = self.create_dict_from_string(section[key])
-                from psyclone.dynamo0p3 import SUPPORTED_FORTRAN_DATATYPES as \
-                    dynamo0p3_datatypes
-                # Check for valid datatypes (filter to remove empty values)
-                datatypes = list(filter(None, all_kinds.keys()))
-                if datatypes != dynamo0p3_datatypes:
-                    raise ConfigurationError(
-                        "Invalid datatype found in the '[dynamo0.3]' "
-                        "section of the configuration file '{0}'. "
-                        "Valid datatypes are: '{1}'."
-                        .format(config.filename, dynamo0p3_datatypes))
-                # Check for valid kinds (filter to remove any empty values)
-                datakinds = list(filter(None, all_kinds.values()))
-                if len(datakinds) != len(dynamo0p3_datatypes):
-                    raise ConfigurationError(
-                        "Did not find default kind for one or more "
-                        "datatypes in the '[dynamo0.3]' section "
-                        "of the configuration file '{0}'."
-                        .format(config.filename))
-                self._default_kind = all_kinds
-            else:
+        # Parse setting for redundant computation over annexed dofs
+        try:
+            self._compute_annexed_dofs = section.getboolean(
+                'COMPUTE_ANNEXED_DOFS')
+        except ValueError as err:
+            raise ConfigurationError(
+                "error while parsing COMPUTE_ANNEXED_DOFS in the [dynamo0.3] "
+                "section of the config file: {0}".format(str(err)),
+                config=self._config)
+
+        # Parse setting for default kinds (precisions)
+        if "default_kind" in section:
+            # Set default kinds (precisions) from config file
+            all_kinds = self.create_dict_from_string(section["default_kind"])
+            from psyclone.dynamo0p3 import SUPPORTED_FORTRAN_DATATYPES
+            # Check for valid datatypes (filter to remove empty values)
+            datatypes = list(filter(None, all_kinds.keys()))
+            if datatypes != SUPPORTED_FORTRAN_DATATYPES:
```

Comment:
I realised this time around that this check is relying on the ordering of the entries in the config file. Therefore if I change that order, e.g.:

    default_kind = logical: l_def: real: r_def, integer: i_def

then it falls over. I suggest doing the comparison using sets instead.

Reply:
Good point. Modified the comparison to use sets.